airflow/providers/amazon/aws/operators/appflow.py

```diff
@@ -0,0 +1,545 @@
+    ) -> None:
+        if source not in {"salesforce", "zendesk"}:
+            raise AirflowException(f"Source {source} is not supported for AppflowRunOperator!")
+        super().__init__(
+            source=source,
+            name=name,
+            flow_update=False,
+            source_field=None,
+            dt=None,
+            poll_interval=poll_interval,
+            aws_conn_id=aws_conn_id,
+            region=region,
+            **kwargs,
+        )
+
+
+class AppflowRunFullOperator(AppflowBaseOperator):
+    """
+    Execute a Appflow full run removing any filter.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AppflowRunFullOperator`
+
+    :param source: The source name (e.g. salesforce, zendesk)
+    :param name: The flow name
+    :param poll_interval: how often in seconds to check the query status
+    :param aws_conn_id: aws connection to use
+    :param region: aws region to use
+    """
+
+    def __init__(
+        self,
+        source: str,
+        name: str,
+        poll_interval: int = 20,
+        aws_conn_id: Optional[str] = "aws_default",
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        if source not in {"salesforce", "zendesk"}:
+            raise AirflowException(f"Source {source} is not supported for AppflowRunFullOperator!")
+        super().__init__(
+            source=source,
+            name=name,
+            flow_update=True,
+            source_field=None,
+            dt=None,
+            poll_interval=poll_interval,
+            aws_conn_id=aws_conn_id,
+            region=region,
+            **kwargs,
+        )
+
+
+class AppflowRunBeforeOperator(AppflowBaseOperator):
+    """
+    Execute a Appflow run after updating the filters to select only previous data.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AppflowRunBeforeOperator`
+
+    :param source: The source name (e.g. salesforce)
+    :param name: The flow name
+    :param source_field: The field name to apply filters
+    :param dt: The date value (or template) to be used in filters.
+    :param poll_interval: how often in seconds to check the query status
+    :param aws_conn_id: aws connection to use
+    :param region: aws region to use
+    """
+
+    template_fields = ("dt",)
+
+    def __init__(
+        self,
+        source: str,
+        name: str,
+        source_field: str,
+        dt: str,
+        poll_interval: int = 20,
+        aws_conn_id: Optional[str] = "aws_default",
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        if not dt:
+            raise AirflowException("The dt argument is mandatory for AppflowRunBeforeOperator!")
+        if source not in {"salesforce"}:
+            raise AirflowException(f"Source {source} is not supported for AppflowRunBeforeOperator!")
```

Comment:
```suggestion
        if source != "salesforce":
            raise ValueError(f"Source {source!r} is not supported for AppflowRunBeforeOperator!")
```
Now that I think of it, perhaps it would be useful to add what sources are supported to the docstring in these operators. This would explicitly expose the information in the Python API documentation for users. WDYT?

Reply:
Done!